### PR TITLE
Port old header and update style

### PR DESCRIPTION
**Pull request by youngkidwarrior**
Create a home header component that matches the [new style from figma](https://www.figma.com/file/k7ct2aqK1zUvZripOWtW8h/send-landing?type=design&node-id=1167-8450&mode=design&t=cTIAsAOgjt0Xw6tQ-4)

Not sure whether this was supposed to be wagmi connector or filled from backend. The current app doesn't allow the user to change wallets. Do we plan to have that feature?
Screenshot 2023-12-22 at 11 47 18 PM

### CODE REVIEW

**Comment by youngkidwarrior** [comment]
@0xBigBoss I modified the `PointsCount` component to be based on referrals, which is what the ⭐️ represents according to Ethen.

Would appreciate a review on the changes I made to the `useUser` hook. Still familiarizing myself with the backend, so not sure if this is the optimal or intended way to query referrals. Might benefit us to store a `totalCount` field for referrals, since this number will probably be used a lot

File: packages/app/components/HomeHeader.tsx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+        size="$2"
+        icon={<Copy />}
+        // @ts-expect-error tamagui doesn't support this yet
+        type="button"
+        onPress={() => {
+          if (user?.profile?.referral_code) {
+            try {
+              // write the referral link to clipboard
+              navigator.clipboard.writeText(referralHref)
```
Comment (0xBigBoss) [inline packages/app/components/HomeHeader.tsx:56]: this is fine for web, but we need a TODO or need a cross platform way of writing to the clipboard.

File: packages/app/utils/useUser.ts [hunk 2]
```diff
@@ -61,16 +61,42 @@ export const useUser = () => {
...
   })()
 
+  const {
+    data: referrals,
+    isLoading: isLoadingReferrals,
+    refetch: refetchReferrals,
+  } = useQuery(['referrals'], {
+    queryFn: async () => {
+      if (!user?.id) return null
+      const { data, error } = await supabase.from('referrals').select('*').eq('referrer_id', user.id)
```
Comment (0xBigBoss) [inline packages/app/utils/useUser.ts:71]: This won't work due to RLS being active and no policies created. I did this so we don't leak other user ids and their tags inadvertently. 

We should create an tRPC API route or create a supabase function to return the data we need.

http://localhost:54323/project/default/auth/policies?search=17663

Screenshot 2024-01-06 at 13 56 27

Comment (0xBigBoss) [reply packages/app/utils/useUser.ts:71]: https://github.com/0xsend/sendapp/blob/ykw/distribution-screen-styles/supabase/migrations/20230827232013_create_referrals_table.sql#L1-L5

Comment (youngkidwarrior) [reply packages/app/utils/useUser.ts:71]: Good to know. Will dig into this

**Review by 0xBigBoss — COMMENTED** [review]
Can we adapt it with a sensible min height as well?

Screenshot 2024-01-11 at 10 56 21

**Comment by youngkidwarrior** [comment]
@0xBigBoss I'm happy enough with this to merge. Other improvements for gradient, colors, connect button, points card, etc can be done in future PRs

**Review by 0xBigBoss — APPROVED** [review]
Screenshot 2024-01-11 at 14 08 44
sounds good @youngkidwarrior could u fix the build?

**Comment by youngkidwarrior** [comment]
@0xBigBoss 

Do you know what's going on here?
```
 FAIL features/home/screen.test.tsx
     app:test │   ● Test suite failed to run
     app:test │ 
     app:test │     Failed to get localhost. Please point to your production server.
     app:test │ 
     app:test │        8 |   localhost = debuggerHost?.split(':')[0]
     app:test │        9 |   if (!localhost) {
     app:test │     > 10 |     throw new Error('Failed to get localhost. Please point to your production server.')
     app:test │          |           ^
     app:test │       11 |   }
     app:test │       12 |   return localhost
```

**Comment by 0xBigBoss** [comment]
@youngkidwarrior the jest tests need mocks. That home screen requires some providers which I haven't had time to set up properly for our unit tests. You can see it only has the `TamaguiProvider`.

https://github.com/0xsend/sendapp/blob/100de48bd07808cd52ccd05635ccc9b80d3d6ce9/packages/app/features/home/screen.test.tsx#L6-L13

If you mock out the data fetching you added, it should start working.
https://github.com/0xsend/sendapp/blob/100de48bd07808cd52ccd05635ccc9b80d3d6ce9/packages/app/features/onboarding/screen.test.tsx#L68-L75

You can test it locally with the `app:test` local resource in tilt or by cd-ing to the app package and runing `yarn test --watch`

**Comment by youngkidwarrior** [comment]
@0xBigBoss Made it through a few mocks, but been stuck on these for about 15-20 minutes. Any help mocking `<Link/>` and `NavigationContainer` from `solito`?

```
 ● HomeScreen

    Couldn't find a navigation object. Is your component inside NavigationContainer?
```

```
    console.error
      The above error occurred in the <LinkCore> component:
      
          at children (/Users/vict0xr/Documents/Send/sendapp/node_modules/solito/src/link/core.tsx:12:3)
```

**Comment by 0xBigBoss** [comment]
> @0xBigBoss Made it through a few mocks, but been stuck on these for about 15-20 minutes. Any help mocking `<Link/>` and `NavigationContainer` from `solito`?
> 
> ```
>  ● HomeScreen
> 
>     Couldn't find a navigation object. Is your component inside NavigationContainer?
> ```
> 
> ```
>     console.error
>       The above error occurred in the <LinkCore> component:
>       
>           at children (/Users/vict0xr/Documents/Send/sendapp/node_modules/solito/src/link/core.tsx:12:3)
> ```

What about this? Though we will need to figure out how to integrate react native navigation properly.

```ts
jest.mock('solito/link', () => ({
  Link: jest.fn(),
}))
```

**Comment by youngkidwarrior** [comment]
Current dependencies on/for this PR:
* `main`
  * **PR #41** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/41?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈
    * **PR #59** <a href="https://app.graphite.dev/github/pr/0xsend/sendapp/59?utm_source=stack-comment-icon" target="_blank">Graphite</a> 

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/0xsend/sendapp/41?utm_source=stack-comment).

**Pull request merged**